### PR TITLE
NetCDF4: various updates and fixes

### PR DESCRIPTION
**Pull request by petersteneteg**

### CODE REVIEW

File: misc/netcdf/python/genericnetcdfsource.py [hunk 1]
```diff
@@ -1,193 +1,204 @@
...
+
+    def collapsed(self, dim: Dimension):
+        minmax = self.minmax(dim)
+        return minmax.x == minmax.y
+
+    def addMinmaxProperty(self, dim: Dimension):
+        clean = self.cleanName(dim.name)
+        if clean not in self.dimensions.properties:
+            minmax = IntMinMaxProperty(clean, dim.name, 0, len(dim) - 1, 0, len(dim) - 1)
+            minmax.visible = False
```
Comment (ResearchDaniel) [inline misc/netcdf/python/genericnetcdfsource.py:59]: Why invisible? Might deserve a comment

File: misc/netcdf/python/genericnetcdfsource.py [hunk 2]
```diff
@@ -1,193 +1,204 @@
...
+
+    def addMinMaxProperties(self, dims: list[Dimension]):
+        for dim in dims:
+            self.addMinmaxProperty(dim)
+
+    def addVariablePropery(self, variable: Variable):
+        enabled = BoolCompositeProperty(self.cleanName(variable.name), variable.name, False)
+        enabled.readOnly = len(variable.shape) < self.outputDimension
+        enabled.collapsed = True
+        dims = StringProperty("dimensions", "Dimensionss",
```
Comment (ResearchDaniel) [inline misc/netcdf/python/genericnetcdfsource.py:70]: Typo, dimensionss

File: misc/netcdf/python/genericnetcdfsource.py [hunk 3]
```diff
@@ -1,193 +1,204 @@
...
+
+            if self.triggerLoad.isModified or first:
+                if len(nonZeroDims) == 0:
+                    self.log("No variables selected")
+                    return
+
+                data = []
+                for var in request:
+                    ranges = [self.minmax(dim) for dim in var.get_dims()]
+                    sizeDims = [r.y - r.x + 1 for r in ranges if r.x != r.y]
```
Comment (ResearchDaniel) [inline misc/netcdf/python/genericnetcdfsource.py:165]: Please add a comment here. What does r.y - r.x + 1 for the dimensions mean?

File: misc/netcdf/python/genericnetcdfsource.py [hunk 4]
```diff
@@ -1,193 +1,204 @@
...
+                    dims = [slice(item.x, item.y + 1) for item in ranges]
+                    varData = var[tuple(dims)]
+                    buffer = numpy.array(varData).astype(
+                        'float32' if self.toFloat.value else var.datatype)
+                    buffer.shape = numpy.flip([1] + sizeDims)
```
Comment (ResearchDaniel) [inline misc/netcdf/python/genericnetcdfsource.py:170]: Why flip? Add comment

File: misc/netcdf/python/processors/netcdfimagesource.py [hunk 5]
```diff
@@ -34,13 +25,8 @@ def processorInfo():
...
 
-    def reloadData(self):
-        extents = []
-        GenericNetCDFSource.reloadData(self, extents)
-
-        buffer = numpy.concatenate(self.data, axis=2)
-        minVal = numpy.amin(buffer)
-        maxVal = numpy.amax(buffer)
+    def dataLoaded(self, data, extents):
+        buffer = numpy.concatenate(data, axis=2)
```
Comment (ResearchDaniel) [inline misc/netcdf/python/processors/netcdfimagesource.py:29]: Is the concatenation used to put things in contiguous memory? Could add a comment to explain

**Review by ResearchDaniel — APPROVED** [review]
Found a typo. Otherwise mostly comments about code that was already there. Would be nice with some clarifications, but not necessary

File: misc/netcdf/python/genericnetcdfsource.py (hunk 4 above)
Comment (petersteneteg) [reply misc/netcdf/python/genericnetcdfsource.py:170]: you have to ask @AnkeAnke

File: misc/netcdf/python/processors/netcdfimagesource.py (hunk 5 above)
Comment (petersteneteg) [reply misc/netcdf/python/processors/netcdfimagesource.py:29]: you have to ask @AnkeAnke

**Pull request merged**